cypress/e2e/basket.spec.cy.js

```diff
@@ -0,0 +1,35 @@
+import BaseToolbarModule from '../pageObjects/modules/base.toolbar.module'
+import ProductsPage from '../pageObjects/products.page'
+import BasketPage from '../pageObjects/basket.page'
+
+describe('Basket tests', function () {
+  beforeEach(function () {
+    cy.loginViaApi(Cypress.env('username'), Cypress.env('password'))
+    cy.visit('/#/search')
+    cy.closeWelcomeBanner()
```

Comment:
As I can see all the page URLs starting from /#/pageName(like '#/search'). And you are using  this approach to login and move/visit page in all the tests.
So as I can see we can optimize those steps in the way creating method for e.g. openSearchPage() in the SearchPage and move two lines to that method
cy.visit('/#/search')
cy.closeWelcomeBanner()
Or the second approach is to create a openPage(pageName) method on commands file to use it to move 
openPage(pageName){
    cy.visit(`/#/${pageName}`)
    cy.closeWelcomeBanner()
}